airflow/providers/google/marketing_platform/hooks/google_ads.py

```diff
@@ -0,0 +1,151 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains Google Ad hook.
+"""
+from tempfile import NamedTemporaryFile
+from typing import IO, Any, Dict, Generator, List
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+from google.ads.google_ads.v2.types import GoogleAdsRow
+from google.api_core.page_iterator import GRPCIterator
+from google.auth.exceptions import GoogleAuthError
+
+from airflow import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+
+class GoogleAdsHook(BaseHook):
+    """
+    Hook for the Google Ads API
+
+    :param gcp_conn_id: The connection ID with the service account details.
+    :type gcp_conn_id: str
+    :param google_ads_conn_id: The connection ID with the details of Google Ads config.yaml file.
+    :type google_ads_conn_id: str
+
+    :return: list of Google Ads Row object(s)
+    :rtype: list[GoogleAdsRow]
+    """
+
+    def __init__(
+        self,
+        gcp_conn_id: str = "google_cloud_default",
+        google_ads_conn_id: str = "google_ads_default",
+    ) -> None:
+        self.gcp_conn_id = gcp_conn_id
+        self.google_ads_conn_id = google_ads_conn_id
+        self.gcp_conn_id = gcp_conn_id
+        self.google_ads_config: Dict[str, Any] = {}
+
+    def _get_service(self) -> GoogleAdsClient:
+        """Connects and authenticates with the Google Ads API using a service account"""
+
+        try:
+            with NamedTemporaryFile("w", suffix=".json") as secrets_temp:
+                self._get_config()
+                self._update_config_with_secret(secrets_temp)
+                client = GoogleAdsClient.load_from_dict(self.google_ads_config)
+                return client.get_service("GoogleAdsService", version="v2")
```

Comment:
Should we make the `version` parameter configurable on the hook level?

Reply:
Good point! I will update.